src/MerchantCenter/PolicyComplianceCheck.php

```diff
@@ -68,6 +70,116 @@ public function is_accessible(): bool {
 		return true;
 	}
 
+	/**
+	 * Check if the store sample product landing pages lead to a 404 error.
+	 *
+	 * @return bool
+	 */
+	public function has_page_not_found_error(): bool {
+		$headers = $this->get_landing_page_headers();
+		if ( ! empty( $headers ) && $headers[0] !== 'HTTP/1.1 200 OK' ) {
+			return true;
+		}
+		return false;
+	}
+
+	/**
+	 * Check if the store sample product landing pages has redirects through 3P domains.
+	 *
+	 * @return bool
+	 */
+	public function has_redirects(): bool {
+		$headers = $this->get_landing_page_headers();
+		if ( ! empty( $headesr ) && isset( $headers[0] ) ) {
+			if ( $headers[0] === 'HTTP/1.1 302 Found' ) {
+				// this is the URL where it's redirecting
+				return true;
+			}
+		}
+			return false;
+	}
+
+	/**
+	 * Get the headers of the store sample product landing pages.
+	 *
+	 * @return array
+	 */
+	public function get_landing_page_headers(): array {
+		$products = wc_get_products(
+			[
+				'limit' => 1,
+			]
+		);
+		if ( ! empty( $products ) ) {
+			$response = wp_remote_get( $products[0]->get_permalink() );
+			if ( is_array( $response ) && ! is_wp_error( $response ) ) {
+				return $response['headers'];
+			}
+		}
+		return [];
+	}
+
+	/**
+	 * Check if the merchant set the restrictions in robots.txt or not in the store.
+	 *
+	 * @return bool
+	 */
+	public function has_restriction(): bool {
+		return ! $this->robots_allowed( $this->get_site_url() );
+	}
+
+	/**
+	 * Check if the robots.txt has restrictions or not in the store.
+	 *
+	 * @param string $url
+	 * @return bool
+	 */
+	private function robots_allowed( $url ) {
+		$agents = [ preg_quote( '*', '/' ) ];
+		$agents = implode( '|', $agents );
+
+		// location of robots.txt file
+		$robotstxt = file( trailingslashit( $url ) . 'robots.txt' );
+
+		// if there isn't a robots, then we're allowed in
+		if ( empty( $robotstxt ) ) {
+			return true;
+		}
+
+		$rules        = [];
+		$rule_applies = false;
+		foreach ( $robotstxt as $line ) {
+			// skip blank lines
+			$line = trim( $line );
+			if ( ! $line ) {
+				continue;
+			}
+
+			// following rules only apply if User-agent matches '*'
+			if ( preg_match( '/^\s*User-agent:\s*(.*)/i', $line, $match ) ) {
+				$rule_applies = preg_match( "/($agents)/i", $match[1] );
```

Comment:
If we are not going to pass agents to the function and are going to only use compare to agent *, then the two lines above to escape and implode become unnecessary and we can just do a simple comparison here:

```suggestion
				$rule_applies = '*' === $match[1];
```